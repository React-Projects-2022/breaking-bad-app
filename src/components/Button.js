import "./../styles/components.css";
import PropTypes from "prop-types";

const Button = ({
  label,
  param,
  action,
  btnClass = "btn btn-info btn-lg btn-block custom-button",
}) => {
  return (
    <>
      <button className={btnClass} onClick={() => action(param)}>
        {label}
      </button>
    </>
  );
};

// Para obligar pasar el argumento para el objeto
Button.propTypes = {
  label: PropTypes.string.isRequired,
};

export default Button;
