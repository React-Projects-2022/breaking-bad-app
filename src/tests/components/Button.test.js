import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Button from "../../components/Button";

describe("Pruebas en Button", () => {
  const label = "Anartz";
  const wrapper = shallow(<Button label={label} />);

  test("Debe de mostrar <Button /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobar que texto del label es correcto", () => {
    expect(wrapper.text()).toBe(label);
  });

  test("Comprobar que las clases aÃ±adidas coinciden con lo establecido", () => {
    const className = "btn btn-info btn-lg btn-block custom-button";
    expect(wrapper.find("button").props().className).toBe(className);
  });
});
