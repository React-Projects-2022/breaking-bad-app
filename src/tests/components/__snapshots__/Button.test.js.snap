// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Button Debe de mostrar <Button /> correctamente 1`] = `
<Fragment>
  <button
    className="btn btn-info btn-lg btn-block custom-button"
    onClick={[Function]}
  >
    Anartz
  </button>
</Fragment>
`;
